### PR TITLE
feat(feedback): Poll for new feedbacks, and show a message so people can load them in

### LINKED ISSUE #61474: [User Feedback] There should be a button/way to refresh the list of feedbacks when the page has been open for a while
Like a button to refresh the list, and/or a background fetch that looks for new data, the reveals an alert at the top of the list: "new data arrived, click to load". That kind of thing.

### PR DESCRIPTION
**Pull request by ryan953**
This PR adds code that polls in the background, every 30second when the feedback page is open, to see if there are newer feedbacks available.

Polling disables itself once it discovers that one new item is available (we're not asking for a count of new things). 
Polling won't start if the user picks an absolute date range.

The purple button takes up it's own row. So when it first appears the list will get a little shorter, and the list items will not scroll 'behind' it, no overlapping.

Clicking the button will reset the `listHeadTime` timestamp, then evict the list cache. Using the new `listHeadTime` a new 1st page will be fetched. If you had previously scrolled down a bunch and loaded in all that data.. it will be re-fetched as you scroll down again.

SCR-20240119-mqvb

Fixes https://github.com/getsentry/sentry/issues/61474

### CODE REVIEW

**Comment by billyvg** [comment]
Can you add some screenshots of this in action?

**Comment by ryan953** [comment]
> Can you add some screenshots of this in action?

working on them

File: static/app/components/feedback/useFeedbackListQueryKey.tsx [hunk 1]
```diff
@@ -38,38 +44,47 @@ export default function useFeedbackListQueryKey({organization}: Props): ApiQuery
...
-          ...rest,
-          start: new Date(now - statsPeriodMs).toISOString(),
-          end: new Date(now).toISOString(),
-        }
-      : // The issues endpoint cannot handle when statsPeroid has a value of "", so
-        // we remove that from the rest and do not use it to query.
-        rest;
-  }, [queryView]);
+
+    if (prefetch) {
```
Comment (michellewzhang) [inline static/app/components/feedback/useFeedbackListQueryKey.tsx:51]: what is the `prefetch` doing? / why do we need it?

**Comment by billyvg** [comment]
imo this would be better as a button, the alert doesn't look like it's clickable, and I think some folks may glaze over it considering the number of Alerts we have all over the place.

File: static/app/components/feedback/useFeedbackListQueryKey.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/components/feedback/useFeedbackListQueryKey.tsx:56]: added a comment about it; It's not the best name

**Comment by ryan953** [comment]
> imo this would be better as a button

@billyvg Switched to a button!

File: static/app/components/feedback/useFeedbackHasNewItems.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import {useEffect, useState} from 'react';
+
+import {ApiQueryKey, useApiQuery} from 'sentry/utils/queryClient';
+
+interface Props {
+  listPrefetchQueryKey: ApiQueryKey | undefined;
+}
+
+const POLLING_INTERVAL_MS = 10 * 1000;
```
Comment (billyvg) [inline static/app/components/feedback/useFeedbackHasNewItems.tsx:9]:
```suggestion
const POLLING_INTERVAL_MS = 10_000;
```

File: static/app/components/feedback/useFeedbackListQueryKey.tsx [hunk 3]
```diff
@@ -38,38 +44,58 @@ export default function useFeedbackListQueryKey({organization}: Props): ApiQuery
...
+    // Note: The ApiQueryKey that we return isn't actually for a full page of
+    // prefetched data, it's just one row actually.
+    if (prefetch) {
+      if (!statsPeriod) {
+        // We shouldn't prefetch if the query uses an absolute date range
+        return undefined;
+      }
+      // Look 1 day into the future, from the time the page is loaded for new
+      // feedbacks to come in.
+      const intervalMS = intervalToMilliseconds('1d');
```
Comment (billyvg) [inline static/app/components/feedback/useFeedbackListQueryKey.tsx:63]: l: Since this is constant we can define the interval instead of calculating it

**Review by billyvg — APPROVED** [review]
Are any of these hooks tested? Can we follow-up with tests?

File: static/app/components/feedback/useFeedbackListQueryKey.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/components/feedback/useFeedbackListQueryKey.tsx:57]: taking suggestions though!

Comment (michellewzhang) [reply static/app/components/feedback/useFeedbackListQueryKey.tsx:57]: so `prefetch` refers to fetching into the future? maybe we could rename it to `futureFetch` or something similar?

**Pull request merged**